drivers/firmware/scmi/power.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/firmware/scmi/power.h>
+#include <string.h>
+
+DT_SCMI_PROTOCOL_DEFINE_NODEV(DT_INST(0, arm_scmi_power), NULL);
```

Comment:
A note on this: not exactly ideal to put this here as we'd normally have a generic power domain driver using this, but since Zephyr doesn't really support DT nodes providing multiple PDs it becomes a bit tricky. For now, I'd say it's fine to go with this.

Reply:
Okay, thanks:)